src/quic/node_quic_util.h

```diff
@@ -73,7 +73,7 @@ enum QuicErrorFamily : int32_t {
 
 // StatsBase is a utility help for classes (like QuicSession)
 // that record performance statistics
-template <typename T>
+template <typename T, typename Q>
```

Comment:
`Q` is currently always an `AsyncWrap` subclass, and that’s the only property of it that is being used – maybe this doesn’t need to be a template parameter?

Reply:
Thought of that also, but the call to Debug() in ~StatsDebug doesn't like passing either `this` or `reinterpret_cast<AsyncWrap*>` as the first argument. I'm liking missing something simple.

Reply:
Hm yeah… I hadn’t thought of that. CRTP is the typical approach for that – I think we could get rid of the double templating by inferring `T` from `Q`, similar to the traits approach used for the header refcount classes, but again, nothing important that needs to be addressed now

Reply:
Went ahead and tweaked it. Using a Traits approach now. Still likely could use some tweak but hopefully much better.